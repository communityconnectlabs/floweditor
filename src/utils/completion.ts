import { AssetStore, CompletionOption } from 'store/flowContext';

export interface CompletionAssets {
  assetStore: AssetStore;
  schema: CompletionSchema;
}

export interface CompletionProperty {
  key: string;
  help: string;
  type: string;
}

export interface CompletionType {
  name: string;

  key_source?: string;
  property_template?: CompletionProperty;
  properties?: CompletionProperty[];
}

export interface CompletionSchema {
  types: CompletionType[];
  root: CompletionProperty[];
}

export const getFunctions = (functions: CompletionOption[], query: string): CompletionOption[] => {
  return functions.filter((option: CompletionOption) => {
    if (option.signature) {
      return option.signature.indexOf(query) === 0;
    }
    return false;
  });
};

/**
 * Takes a dot query and returns the completions options at the current level
 * @param dotQuery query such as "results.field_name.ca"
 */
export const getCompletions = (
  completion: CompletionAssets,
  dotQuery: string
): CompletionOption[] => {
  const parts = (dotQuery || '').split('.');
  let currentProps: CompletionProperty[] = completion.schema.root;

  let prefix = '';
  let part = '';
  while (parts.length > 0) {
    part = parts.shift();
    if (part) {
      // eslint-disable-next-line
      const nextProp = currentProps.find((prop: CompletionProperty) => prop.key === part);
      if (nextProp) {
        // eslint-disable-next-line
        const nextType = completion.schema.types.find(
          (type: CompletionType) => type.name === nextProp.type
        );
        if (nextType && nextType.properties) {
          currentProps = nextType.properties;
          prefix += part + '.';
        } else if (nextType && nextType.property_template) {
          prefix += part + '.';
          const template = nextType.property_template;
          currentProps = Object.keys(completion.assetStore[nextType.name].items).map(
            (key: string) => ({
              key: template.key.replace('{key}', key),
              help: template.help.replace('{key}', key),
              type: template.type
            })
          );
        } else {
          // eslint-disable-next-line
          currentProps = currentProps.filter((prop: CompletionProperty) =>
            prop.key.startsWith(part.toLowerCase())
          );
          break;
        }
      } else {
        // eslint-disable-next-line
        currentProps = currentProps.filter((prop: CompletionProperty) =>
          prop.key.startsWith(part.toLowerCase())
        );
        break;
      }
    }
  }

  return currentProps.map((prop: CompletionProperty) => {
    const name =
      prop.key === '__default__' ? prefix.substr(0, prefix.length - 1) : prefix + prop.key;
    return { name, summary: prop.help };
  });
};

export const getCompletionName = (option: CompletionOption): string => {
  return option.name || option.signature.substr(0, option.signature.indexOf('('));
};

export const getCompletionSignature = (option: CompletionOption): string => {
  return option.signature.substr(option.signature.indexOf('('));
};
