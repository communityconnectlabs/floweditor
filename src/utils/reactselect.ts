import { StylesConfig } from 'react-select';

export const large = {
  placeholder: (styles: StylesConfig, state: any) => {
    return { ...styles, color: '#ddd', marginLeft: '3px' };
  },
  singleValue: (styles: StylesConfig, state: any) => {
    return { ...styles, marginLeft: '3px' };
  },
  multiValue: (styles: StylesConfig, state: any) => {
    return { ...styles, marginLeft: '0px' };
  },
  input: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      marginLeft: '0px',
      caretColor: '#999',
      marginBottom: '0px',
      boxShadow: 'none',
      border: 'none'
    };
  },
  control: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      borderColor: '#ccc',
      boxShadow: 'none',
      transition: 'all 0.3s ease-in-out',
      '&:hover': {
        borderColor: '#ccc'
      }
    };
  }
};

export const messageStyle = {
  control: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      borderColor: '#ccc',
      boxShadow: 'none',
      borderBottom: 'none',
      borderBottomRightRadius: '0px',
      borderBottomLeftRadius: '0px',
      '&:hover': {
        borderColor: '#ccc'
      }
    };
  }
};

export const getErroredSelect = (baseControl: any): any => {
  return {
    control: (styles: StylesConfig) => ({
      ...styles,
      ...baseControl,
      borderColor: 'tomato',
      boxShadow: '0 0 0 3px rgba(255,196,186,0.5) !important;',
      '&:hover': {
        borderColor: 'tomato'
      }
    })
  };
};

export const small = {
  control: (styles: StylesConfig) => ({
    ...styles,
    borderColor: '#ccc',
    boxShadow: 'none',
    '&:hover': {
      borderColor: '#ccc'
    },
    maxHeight: '28px',
    minHeight: '28px',
    fontSize: '12px',
    lineHeight: '12px'
  }),
  menuList: (styles: StylesConfig) => ({
    ...styles,
    maxHeight: 200
  }),
  option: (styles: StylesConfig) => ({
    ...styles,
    fontSize: 12
  }),
  singleValue: (styles: StylesConfig) => ({
    ...styles,
    maxHeight: 28,
    lineHeight: '13px'
  }),
  indicatorsContainer: (styles: StylesConfig) => ({
    ...styles,
    height: 28,
    maxHeight: 28
  }),
  valueContainer: (styles: StylesConfig) => ({
    ...styles,
    height: 28,
    maxHeight: 28
  }),
  input: (styles: StylesConfig) => ({
    ...styles,
    marginLeft: '-2px',
    marginTop: '-10px',
    caretColor: '#999',
    boxShadow: 'none',
    border: 'none',
    maxHeight: '20px',
    minHeight: '20px',
    lineHeight: '12px'
  })
};

export const tagging = {
  control: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      borderColor: '#ccc',
      boxShadow: 'none',
      transition: 'all 0.3s ease-in-out',
      '&:hover': {
        borderColor: '#ccc'
      }
    };
  },
  input: (styles: StylesConfig, state: any) => {
    return {
      ...styles,
      marginLeft: '0px',
      caretColor: '#999',
      marginBottom: '0px',
      boxShadow: 'none',
      border: 'none'
    };
  },
  multiValue: (styles: StylesConfig) => {
    return {
      ...styles,
      maxHeight: 24,
      minHeight: 24,
      lineHeight: '17px',
      backgroundColor: 'hsl(0, 0%, 94%)',
      border: '1px solid hsl(0, 0%, 85%)'
    };
  }
};
