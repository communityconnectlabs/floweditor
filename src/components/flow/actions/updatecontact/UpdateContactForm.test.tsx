import UpdateContactForm from 'components/flow/actions/updatecontact/UpdateContactForm';
import { mock } from 'testUtils';
import {
  createSetContactFieldAction,
  getActionFormProps,
  createSetContactChannelAction,
  createSetContactNameAction,
  createSetContactLanguageAction
} from 'testUtils/assetCreators';
import * as utils from 'utils';
import * as React from 'react';
import { render, fireEvent } from 'test/utils';

mock(utils, 'createUUID', utils.seededUUIDs());

// These are the RTL-based tests
describe(UpdateContactForm.name, () => {
  it('should render field', () => {
    const { baseElement } = render(
      <UpdateContactForm {...getActionFormProps(createSetContactFieldAction())} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render channel', () => {
    const { baseElement } = render(
      <UpdateContactForm {...getActionFormProps(createSetContactChannelAction())} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render name', () => {
    const { baseElement } = render(
      <UpdateContactForm {...getActionFormProps(createSetContactNameAction())} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render language', () => {
    const { baseElement } = render(
      <UpdateContactForm {...getActionFormProps(createSetContactLanguageAction())} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('shouldnt allow expressions in language selection', () => {
    const { getAllByDisplayValue } = render(
      <UpdateContactForm {...getActionFormProps(createSetContactLanguageAction())} />
    );

    // we don't support expressions but this shouldn't blow up either
    const input = getAllByDisplayValue('')[1];
    fireEvent.change(input, { target: { value: '@' } });
  });
});
