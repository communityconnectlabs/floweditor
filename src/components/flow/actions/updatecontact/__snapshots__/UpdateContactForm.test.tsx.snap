// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateContactForm should render channel 1`] = `
<body>
  <div>
    <div
      class="dialog"
    >
      <div
        class="header set_contact_field"
      >
        <div
          class="title_container"
        >
          <div
            class="title"
          >
            Update Contact
          </div>
          <div
            class="subtitle"
          />
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class=" type_list"
        >
          <p>
            When a contact arrives at this point in your flow...
          </p>
          <div>
            <select
              data-testid="select"
            >
              <option
                value="send_msg"
              >
                Send the contact a message
              </option>
              <option
                value="wait_for_response"
              >
                Wait for the contact to respond
              </option>
              <option
                value="send_broadcast"
              >
                Send somebody else a message
              </option>
              <option
                value="add_input_labels"
              >
                Label the incoming message
              </option>
              <option
                value="add_contact_urn"
              >
                Add a URN for the contact
              </option>
              <option
                value="add_contact_groups"
              >
                Add the contact to a group
              </option>
              <option
                value="remove_contact_groups"
              >
                Remove the contact from a group
              </option>
              <option
                value="set_contact_field"
              >
                Update the contact
              </option>
              <option
                value="send_email"
              >
                Send an email
              </option>
              <option
                value="set_run_result"
              >
                Save a result for this flow
              </option>
              <option
                value="call_webhook"
              >
                Call a webhook
              </option>
              <option
                value="call_resthook"
              >
                Call Zapier
              </option>
              <option
                value="enter_flow"
              >
                Enter another flow
              </option>
              <option
                value="start_session"
              >
                Start somebody else in a flow
              </option>
              <option
                value="transfer_airtime"
              >
                Send the contact airtime
              </option>
              <option
                value="split_by_intent"
              >
                Split by intent
              </option>
              <option
                value="split_by_expression"
              >
                Split by a custom expression
              </option>
              <option
                value="split_by_contact_field"
              >
                Split by a contact field
              </option>
              <option
                value="split_by_run_result"
              >
                Split by a result in the flow
              </option>
              <option
                value="split_by_random"
              >
                Split by random chance
              </option>
              <option
                value="split_by_groups"
              >
                Split by group membership
              </option>
            </select>
          </div>
        </div>
        <p>
          Select what to update
        </p>
        <div
          class="ele"
        >
          <div
            class="react-select selection css-2b097c-container"
          >
            <div
              class=" css-1h1mxjq-control"
            >
              <div
                class=" css-1hwfws3"
              >
                <div
                  class=" css-1sdgdfu-singleValue"
                >
                  Channel
                </div>
                <div
                  class="css-1al1kbh-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-3-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="value"
        >
          <div
            class="ele"
          >
            <div
              class="react-select selection css-2b097c-container"
            >
              <div
                class=" css-1h1mxjq-control"
              >
                <div
                  class=" css-1hwfws3"
                >
                  <div
                    class=" css-1sdgdfu-singleValue"
                  >
                    Twilio Channel
                  </div>
                  <div
                    class="css-1al1kbh-Input"
                  >
                    <div
                      class=""
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="react-select-4-input"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="buttons"
        >
          <div
            class="right_buttons"
          >
            <div
              class="btn secondary  "
              style="margin-left: 0px; margin-top: 0px;"
            >
              Cancel
            </div>
            <div
              class="btn primary  "
              style="margin-left: 10px; margin-top: 0px;"
            >
              Ok
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`UpdateContactForm should render field 1`] = `
<body>
  <div>
    <div
      class="dialog"
    >
      <div
        class="header set_contact_field"
      >
        <div
          class="title_container"
        >
          <div
            class="title"
          >
            Update Contact
          </div>
          <div
            class="subtitle"
          />
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class=" type_list"
        >
          <p>
            When a contact arrives at this point in your flow...
          </p>
          <div>
            <select
              data-testid="select"
            >
              <option
                value="send_msg"
              >
                Send the contact a message
              </option>
              <option
                value="wait_for_response"
              >
                Wait for the contact to respond
              </option>
              <option
                value="send_broadcast"
              >
                Send somebody else a message
              </option>
              <option
                value="add_input_labels"
              >
                Label the incoming message
              </option>
              <option
                value="add_contact_urn"
              >
                Add a URN for the contact
              </option>
              <option
                value="add_contact_groups"
              >
                Add the contact to a group
              </option>
              <option
                value="remove_contact_groups"
              >
                Remove the contact from a group
              </option>
              <option
                value="set_contact_field"
              >
                Update the contact
              </option>
              <option
                value="send_email"
              >
                Send an email
              </option>
              <option
                value="set_run_result"
              >
                Save a result for this flow
              </option>
              <option
                value="call_webhook"
              >
                Call a webhook
              </option>
              <option
                value="call_resthook"
              >
                Call Zapier
              </option>
              <option
                value="enter_flow"
              >
                Enter another flow
              </option>
              <option
                value="start_session"
              >
                Start somebody else in a flow
              </option>
              <option
                value="transfer_airtime"
              >
                Send the contact airtime
              </option>
              <option
                value="split_by_intent"
              >
                Split by intent
              </option>
              <option
                value="split_by_expression"
              >
                Split by a custom expression
              </option>
              <option
                value="split_by_contact_field"
              >
                Split by a contact field
              </option>
              <option
                value="split_by_run_result"
              >
                Split by a result in the flow
              </option>
              <option
                value="split_by_random"
              >
                Split by random chance
              </option>
              <option
                value="split_by_groups"
              >
                Split by group membership
              </option>
            </select>
          </div>
        </div>
        <p>
          Select what to update
        </p>
        <div
          class="ele"
        >
          <div
            class="react-select selection css-2b097c-container"
          >
            <div
              class=" css-1h1mxjq-control"
            >
              <div
                class=" css-1hwfws3"
              >
                <div
                  class=" css-1sdgdfu-singleValue"
                >
                  Age
                </div>
                <div
                  class="css-1al1kbh-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-2-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="value"
        >
          <div
            class="ele"
          >
            <div
              class="wrapper"
            >
              <input
                class="textinput"
                data-spec="input"
                data-testid="input"
                maxlength="-1"
                name="Field Value"
                placeholder="Enter a new value for Age"
                type="text"
                value="25"
              />
              <div
                class="completion_container hidden"
                data-spec="completion-options"
                style="top: 0px; left: 0px;"
              >
                <div
                  class="options_wrapper"
                >
                  <ul
                    class="option_list"
                    data-spec="completion-list"
                  />
                  <div
                    class="help"
                    data-spec="completion-help"
                  >
                    Tab to complete, enter to select
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="buttons"
        >
          <div
            class="right_buttons"
          >
            <div
              class="btn secondary  "
              style="margin-left: 0px; margin-top: 0px;"
            >
              Cancel
            </div>
            <div
              class="btn primary  "
              style="margin-left: 10px; margin-top: 0px;"
            >
              Ok
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`UpdateContactForm should render language 1`] = `
<body>
  <div>
    <div
      class="dialog"
    >
      <div
        class="header set_contact_field"
      >
        <div
          class="title_container"
        >
          <div
            class="title"
          >
            Update Contact
          </div>
          <div
            class="subtitle"
          />
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class=" type_list"
        >
          <p>
            When a contact arrives at this point in your flow...
          </p>
          <div>
            <select
              data-testid="select"
            >
              <option
                value="send_msg"
              >
                Send the contact a message
              </option>
              <option
                value="wait_for_response"
              >
                Wait for the contact to respond
              </option>
              <option
                value="send_broadcast"
              >
                Send somebody else a message
              </option>
              <option
                value="add_input_labels"
              >
                Label the incoming message
              </option>
              <option
                value="add_contact_urn"
              >
                Add a URN for the contact
              </option>
              <option
                value="add_contact_groups"
              >
                Add the contact to a group
              </option>
              <option
                value="remove_contact_groups"
              >
                Remove the contact from a group
              </option>
              <option
                value="set_contact_field"
              >
                Update the contact
              </option>
              <option
                value="send_email"
              >
                Send an email
              </option>
              <option
                value="set_run_result"
              >
                Save a result for this flow
              </option>
              <option
                value="call_webhook"
              >
                Call a webhook
              </option>
              <option
                value="call_resthook"
              >
                Call Zapier
              </option>
              <option
                value="enter_flow"
              >
                Enter another flow
              </option>
              <option
                value="start_session"
              >
                Start somebody else in a flow
              </option>
              <option
                value="transfer_airtime"
              >
                Send the contact airtime
              </option>
              <option
                value="split_by_intent"
              >
                Split by intent
              </option>
              <option
                value="split_by_expression"
              >
                Split by a custom expression
              </option>
              <option
                value="split_by_contact_field"
              >
                Split by a contact field
              </option>
              <option
                value="split_by_run_result"
              >
                Split by a result in the flow
              </option>
              <option
                value="split_by_random"
              >
                Split by random chance
              </option>
              <option
                value="split_by_groups"
              >
                Split by group membership
              </option>
            </select>
          </div>
        </div>
        <p>
          Select what to update
        </p>
        <div
          class="ele"
        >
          <div
            class="react-select selection css-2b097c-container"
          >
            <div
              class=" css-1h1mxjq-control"
            >
              <div
                class=" css-1hwfws3"
              >
                <div
                  class=" css-1sdgdfu-singleValue"
                >
                  Language
                </div>
                <div
                  class="css-1al1kbh-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-6-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="value"
        >
          <div
            class="ele"
          >
            <div
              class="react-select selection css-2b097c-container"
            >
              <div
                class=" css-1h1mxjq-control"
              >
                <div
                  class=" css-1hwfws3"
                >
                  <div
                    class=" css-1sdgdfu-singleValue"
                  >
                    eng
                  </div>
                  <div
                    class="css-1al1kbh-Input"
                  >
                    <div
                      class=""
                      style="display: inline-block;"
                    >
                      <input
                        aria-autocomplete="list"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        id="react-select-7-input"
                        spellcheck="false"
                        style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                      <div
                        style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-6q0nyr-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="buttons"
        >
          <div
            class="right_buttons"
          >
            <div
              class="btn secondary  "
              style="margin-left: 0px; margin-top: 0px;"
            >
              Cancel
            </div>
            <div
              class="btn primary  "
              style="margin-left: 10px; margin-top: 0px;"
            >
              Ok
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`UpdateContactForm should render name 1`] = `
<body>
  <div>
    <div
      class="dialog"
    >
      <div
        class="header set_contact_field"
      >
        <div
          class="title_container"
        >
          <div
            class="title"
          >
            Update Contact
          </div>
          <div
            class="subtitle"
          />
        </div>
      </div>
      <div
        class="content"
      >
        <div
          class=" type_list"
        >
          <p>
            When a contact arrives at this point in your flow...
          </p>
          <div>
            <select
              data-testid="select"
            >
              <option
                value="send_msg"
              >
                Send the contact a message
              </option>
              <option
                value="wait_for_response"
              >
                Wait for the contact to respond
              </option>
              <option
                value="send_broadcast"
              >
                Send somebody else a message
              </option>
              <option
                value="add_input_labels"
              >
                Label the incoming message
              </option>
              <option
                value="add_contact_urn"
              >
                Add a URN for the contact
              </option>
              <option
                value="add_contact_groups"
              >
                Add the contact to a group
              </option>
              <option
                value="remove_contact_groups"
              >
                Remove the contact from a group
              </option>
              <option
                value="set_contact_field"
              >
                Update the contact
              </option>
              <option
                value="send_email"
              >
                Send an email
              </option>
              <option
                value="set_run_result"
              >
                Save a result for this flow
              </option>
              <option
                value="call_webhook"
              >
                Call a webhook
              </option>
              <option
                value="call_resthook"
              >
                Call Zapier
              </option>
              <option
                value="enter_flow"
              >
                Enter another flow
              </option>
              <option
                value="start_session"
              >
                Start somebody else in a flow
              </option>
              <option
                value="transfer_airtime"
              >
                Send the contact airtime
              </option>
              <option
                value="split_by_intent"
              >
                Split by intent
              </option>
              <option
                value="split_by_expression"
              >
                Split by a custom expression
              </option>
              <option
                value="split_by_contact_field"
              >
                Split by a contact field
              </option>
              <option
                value="split_by_run_result"
              >
                Split by a result in the flow
              </option>
              <option
                value="split_by_random"
              >
                Split by random chance
              </option>
              <option
                value="split_by_groups"
              >
                Split by group membership
              </option>
            </select>
          </div>
        </div>
        <p>
          Select what to update
        </p>
        <div
          class="ele"
        >
          <div
            class="react-select selection css-2b097c-container"
          >
            <div
              class=" css-1h1mxjq-control"
            >
              <div
                class=" css-1hwfws3"
              >
                <div
                  class=" css-1sdgdfu-singleValue"
                >
                  Name
                </div>
                <div
                  class="css-1al1kbh-Input"
                >
                  <div
                    class=""
                    style="display: inline-block;"
                  >
                    <input
                      aria-autocomplete="list"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      id="react-select-5-input"
                      spellcheck="false"
                      style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-6q0nyr-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="value"
        >
          <div
            class="ele"
          >
            <div
              class="wrapper"
            >
              <input
                class="textinput"
                data-spec="input"
                data-testid="input"
                maxlength="-1"
                name="Name"
                placeholder="Enter a new name for the contact"
                type="text"
                value="Jane Goodall"
              />
              <div
                class="completion_container hidden"
                data-spec="completion-options"
                style="top: 0px; left: 0px;"
              >
                <div
                  class="options_wrapper"
                >
                  <ul
                    class="option_list"
                    data-spec="completion-list"
                  />
                  <div
                    class="help"
                    data-spec="completion-help"
                  >
                    Tab to complete, enter to select
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
      >
        <div
          class="buttons"
        >
          <div
            class="right_buttons"
          >
            <div
              class="btn secondary  "
              style="margin-left: 0px; margin-top: 0px;"
            >
              Cancel
            </div>
            <div
              class="btn primary  "
              style="margin-left: 10px; margin-top: 0px;"
            >
              Ok
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
